File: src/main/java/com/senderman/miniroulette/MiniRouletteBot.java
package com.senderman.miniroulette;

import com.annimon.tgbotsmodule.BotHandler;
import com.annimon.tgbotsmodule.BotModule;
import com.annimon.tgbotsmodule.Runner;
import com.annimon.tgbotsmodule.beans.Config;
import io.micronaut.context.event.StartupEvent;
import io.micronaut.runtime.Micronaut;
import io.micronaut.runtime.event.annotation.EventListener;
import jakarta.inject.Singleton;
import org.jetbrains.annotations.NotNull;

import java.util.List;

@Singleton
public class MiniRouletteBot implements BotModule {

    private final BotHandler botHandler;

    public MiniRouletteBot(BotHandler botHandler) {
        this.botHandler = botHandler;
    }

    public static void main(String[] args) {
        Micronaut.build(args)
                .classes(MiniRouletteBot.class)
                .banner(!System.getProperties().containsKey("disableBanner"))
                .start();
    }

    @EventListener
    protected void run(StartupEvent event) {
        Runner.run(List.of(this));
    }

    @Override
    public @NotNull BotHandler botHandler(@NotNull Config config) {
        return botHandler;
    }


}

File: src/main/kotlin/com/senderman/miniroulette/MiniRouletteBot.kt
package com.senderman.miniroulette

import com.annimon.tgbotsmodule.BotHandler
import com.annimon.tgbotsmodule.BotModule
import com.annimon.tgbotsmodule.Runner
import com.annimon.tgbotsmodule.beans.Config
import com.annimon.tgbotsmodule.services.YamlConfigLoaderService

class MiniRouletteBot : BotModule {
    override fun botHandler(config: Config): BotHandler {
        val configLoader = YamlConfigLoaderService<BotConfig>()
        val configFile = configLoader.configFile("/botConfigs/roulette", config.profile)
        val botConfig = configLoader.load(configFile, BotConfig::class.java)
        Services.botConfig = botConfig
        return RouletteBotHandler()
    }
}

fun main(args: Array<String>) {
    val profile = if (args.isNotEmpty() && args[0].isNotBlank()) args[0] else ""
    Runner.run(profile, listOf(MiniRouletteBot()))
}

